BUILD_SHARED_LIBS & POCO_STATIC cause #pragma comment(lib, "PocoUtil" POCO_LIB_SUFFIX) to be invalid

Hello Poco Team,

**Describe the bug**
The visual studio compiler/linker ask for Foundation.lib  instead of Foundation**md**.lib

**To Reproduce**
I build poco from source, as static, with cmake, using BUILD_SHARED_LIBS =OFF. I keep MT=OFF (to use system/compiler dll).
I got files ending with md.lib and mdd.lib => this is what I expected

I add lib and include dependencies to my project (visual studio) using cmake
I add
`#include "Foundation.h" `
in the main

call compiler OK
call linker NOT OK 

**Expected behavior**

The linker shoud ask for PocoFoundationmd.lib, not for :
**Logs**

> LINK : fatal error LNK1104: cannot open file 'PocoFoundation.lib'

**Please add relevant environment information:**
 - Windows 10
 - POCO **13.3**, from what I can see the same problem should apply to the last version (**main**)


**Additional context**
From what I can see it is due to the check performed in Foundation.h 
```
#if defined(_WIN32) && defined(_DLL)
	#if !defined(POCO_DLL) && !defined(POCO_STATIC)
		#define POCO_DLL
	#endif
#endif

```

It cause the following:
`		#pragma comment(lib, "PocoUtil" POCO_LIB_SUFFIX)`
To be invalid (and all similar #pragma) since POCO_LIB_SUFFIX is created using POCO_DLL.

I supposed it should not use POCO_STATIC here because it is deprecated according the cmakefile:
`	message(WARNING "POCO_STATIC has been deprecated. Please use BUILD_SHARED_LIBS=NO to build static libraries.") `
, but use BUILD_SHARED_LIBS  (or a combination of both until complete removal)

BTW, I know the documentation :https://docs.pocoproject.org/current/99150-WindowsPlatformNotes.html
ask to use POCO_STATIC, but I guess the code is supposed to be more up to date. I'm right ?

Sorry if I miss something obvious.

Thanks by advance, 

Have a good day

--- a/CppUnit/CMakeLists.txt
+++ b/CppUnit/CMakeLists.txt
@@ -28,13 +28,6 @@ if(WIN32)
 	endif()
 endif()
 
-if(NOT BUILD_SHARED_LIBS)
-	target_compile_definitions(CppUnit
-		PUBLIC
-			POCO_STATIC
-	)
-elseif(MINGW)
-	target_compile_definitions(CppUnit
-		PUBLIC
-			_DLL)
+if(BUILD_SHARED_LIBS AND MINGW)
+	target_compile_definitions(CppUnit PUBLIC _DLL)
 endif()

--- a/DNSSD/CMakeLists.txt
+++ b/DNSSD/CMakeLists.txt
@@ -39,10 +39,6 @@ POCO_SOURCES_AUTO( SRCS ${SRCS_G})
 file(GLOB_RECURSE HDRS_G "include/*.h" )
 POCO_HEADERS_AUTO( SRCS ${HDRS_G})
 
-if (NOT POCO_STATIC)
-    add_compile_definitions(THREADSAFE)
-endif (NOT POCO_STATIC)
-
 add_library( "${LIBNAME}" ${LIB_MODE} ${SRCS} )
 add_library( "${POCO_LIBNAME}" ALIAS "${LIBNAME}")
 set_target_properties( "${LIBNAME}"
@@ -60,6 +56,10 @@ target_include_directories( "${LIBNAME}"
     PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/src
     )
 target_compile_definitions("${LIBNAME}" PUBLIC ${LIB_MODE_DEFINITIONS})
+
+if (BUILD_SHARED_LIBS)
+	target_compile_definitions(${LIBNAME} PUBLIC THREADSAFE)
+endif ()
 
 POCO_INSTALL("${LIBNAME}")
 POCO_GENERATE_PACKAGE("${LIBNAME}")

--- a/Foundation/CMakeLists.txt
+++ b/Foundation/CMakeLists.txt
@@ -128,13 +128,6 @@ if(NOT DISABLE_CPP14 AND CMAKE_VERSION VERSION_GREATER "3.8")
 	target_compile_features(Foundation PUBLIC cxx_std_14)
 endif()
 
-if(NOT BUILD_SHARED_LIBS)
-	target_compile_definitions(Foundation
-		PUBLIC
-			POCO_STATIC
-	)
-endif()
-
 if(WIN32)
 	target_compile_definitions(Foundation PUBLIC POCO_OS_FAMILY_WINDOWS UNICODE _UNICODE)
 	if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")

--- a/build_cmake.sh
+++ b/build_cmake.sh
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-# POCO_STATIC=1 - for static build
+# BUILD_SHARED_LIBS=OFF - for static build
 # POCO_UNBUNDLED - for no built-in version of libs
 # CMAKE_INSTALL_PREFIX=path - for install path
 

--- a/cmake/DefinePlatformSpecific.cmake
+++ b/cmake/DefinePlatformSpecific.cmake
@@ -22,6 +22,12 @@
 #   - PocoXmdd.lib for /MD debug build
 #   - PocoXmtd.lib for /MT debug build
 
+if(BUILD_SHARED_LIBS)
+	add_compile_definitions(POCO_DLL)
+else()
+	add_compile_definitions(POCO_STATIC)
+endif()
+
 if(MSVC)
 	if(POCO_MT)
 		set(CompilerFlags

